packages/use-midi/src/index.ts

```diff
@@ -67,7 +75,12 @@ const useMidi = (options: Options = {}): Returns => {
       default:
         debugLogger(debug, 'Adding `midiMessage` to `midiMessages` array.', { midiMessage, options })
 
-        setMidiMessages(prevMidiMessages => [...prevMidiMessages, midiMessage])
+        setMidiMessages(prevMidiMessages => {
+          if (prevMidiMessages.length === messageHistoryCount) {
```

Comment:
I'm not sure whether mutating the `prevMidiMessages` is a good idea. Maybe we should do this inline or in a tiny helper function (that copies the array before `shift()`ing it.

Reply:
yeah I was wondering the same thing, I've tested it in local and it worked great, but I totally agree with you that I think we should not mutate this! 👍